pkg/systemd/logs.js

```diff
@@ -288,6 +380,391 @@ $(function() {
+        var pd_t = $('<li>').append($('<a>').append($('<span translatable="yes">').text('Problem details')));
+
+        var ge = $('<div>');
+        var pi = $('<div>');
+        var pd = $('<div class="panel-group" id="accordion-markup">');
+        var tab = $('<ul class="nav nav-tabs nav-tabs-pf">');
+
+        var d_btn =  $('<button class="btn btn-danger problem-btn btn-delete pficon pficon-delete">');
+
+        var r_btn = $();
+        if (problem.IsReported){
+            for (var pid = 0; pid < problem.Reports.length; pid++) {
+                if (problem.Reports[pid][0] === 'ABRT Server') {
+                    var url = problem.Reports[pid][1]['URL']['v']['v'];
+                    r_btn = $('<a class="problem-btn">')
+                            .attr('href', url)
+                            .attr("target", "_blank")
+                            .text('Reported');
+                    break;
+                }
+            }
+        }
+
+        else if (problem.CanBeReported){
+            r_btn = $('<button class="btn btn-primary problem-btn">').text('Report');
+
+            r_btn.click(function() {
+                tab.children(':last-child').replaceWith($('<div class="spinner problem-btn">'));
+                var proc = cockpit.spawn(['reporter-ureport', '-d', problem.ID], { superuser: 'true' });
+                proc.done(function() {
+                    window.location.reload();
+                });
+                proc.fail(function(ex) {
+                    var message;
+                    if (ex.problem === 'access-denied') {
+                        message = _('Not authorized to upload-report');
+                    } else if (ex.problem === "not-found") {
+                        message = _('Reporter "reporter-ureport" not found.');
+                    } else {
+                        message = _('Reporting was unsucessful. Try running `reporter-ureport -d '+ problem.ID +'`');
+                    }
+                    $('<div class="alert alert-danger">')
+                        .append('<span class="pficon pficon-error-circle-o">')
+                        .text(message).insertAfter(".breadcrumb");
+                    tab.children(':last-child').replaceWith($('<span>'));
+                });
+            });
+        }
+
+        ge_t.click(function() {
+            switch_tab(ge_t, ge);
+        });
+
+        pi_t.click(function() {
+            switch_tab(pi_t, pi);
+        });
+
+        pd_t.click(function() {
+            switch_tab(pd_t, pd);
+        });
+
+        d_btn.click(function() {
+            service.DeleteProblems([problem.path]);
+            cockpit.location.go('/');
+        });
+
+        // write into general tab non-ABRT related items
+        var keys = Object.keys(entry).sort();
+        $.each(keys, function(i, key) {
+            if (key !== 'MESSAGE' && !key.startsWith('PROBLEM_')) {
```

Comment:
`startsWith` is unfortunately not portable enough.  You could use `key.indexOf("PROBLEM_") == 0`.  Why do we filter out the PROBLEM_  fields, actually? I think it would be okay to keep them.